src/extensions/scratch3_translate/index.js

```diff
@@ -20,17 +23,27 @@ const serverURL = 'https://translate-service.scratch.mit.edu/';
 const serverTimeoutMs = 10000; // 10 seconds (chosen arbitrarily).
 
 /**
- * Class for the translate-related block in Scratch 3.0
+ * Class for the translate block in Scratch 3.0.
  * @constructor
  */
 class Scratch3TranslateBlocks {
     constructor () {
         /**
-         * List of supported language name and language code pairs.
+         * Language code of the viewer, based on their locale.
+         * @type {string}
+         * @private
+         */
+        this._viewerLanguageCode = this.getViewerLanguageCode();
+
+        /**
+         * List of supported language name and language code pairs, for use in the block menu.
          * @type {Array.<object.<string, string>>}
          * @private
          */
-        this._supportedLanguages = null;
+        this._supportedLanguages = languageNames.menuMap[this._viewerLanguageCode].map(entry => {
```

Comment:
@picklesrus Would it make sense for us to export this format, so we don't need to map?